spec/controllers/api/v1/books_controller_spec.rb

```diff
@@ -0,0 +1,54 @@
+require 'rails_helper'
+
+RSpec.shared_context 'when authenticated User' do
+  let(:user) { create(:user) }
+
+  before do
+    request.headers.merge! user.create_new_auth_token
+  end
+end
+
+describe BooksController, type: :controller do
+  include_context 'when authenticated User'
+
+  describe 'GET #index' do
+    context 'when fetching all the books' do
+      let!(:books) { create_list(:book, 5) }
+
+      before do
+        get :index
```

Comment:
I prefer to declare subject instead of realizing a request in before body

Reply:
Done for both, index and show